### PR TITLE
feat: add search to the create insights page

### LINKED ISSUE #932: Feature: Add search to the create insights page
### Type of feature

🍕 Feature

### Current behavior

Screen Shot 2023-03-01 at 3 22 23 PM

We connected GitHub search to repositories list, but we should add it to the insights creation/edit page.

### Suggested solution

Screen Shot 2023-03-01 at 3 29 03 PM

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by shamimbinnur**
## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Description
This PR adds the functionality to search any repository and add it to the insights page.

## Related Tickets & Documents
Fixes #932 

## Mobile & Desktop Screenshots/Recordings
Screenshot 2023-03-07 at 1 21 39 AM
Screenshot 2023-03-07 at 1 21 29 AM

## Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: components/organisms/InsightPage/InsightPage.tsx [hunk 1]
```diff
@@ -265,11 +288,11 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
           <Title className="!text-1xl !leading-none " level={4}>
             Add Repositories
           </Title>
-
-          <TextInput
-            value={repoToAdd}
-            placeholder="Repository Full Name (ex: open-sauced/open-sauced)"
-            handleChange={handleOnRepoChange}
+
```
Comment (brandonroberts) [inline components/organisms/InsightPage/InsightPage.tsx:291]: nit: remove extra spaces

File: components/organisms/InsightPage/InsightPage.tsx [hunk 2]
```diff
@@ -238,6 +239,28 @@ const InsightPage = ({ edit, insight, pageRepos }: InsightPageProps) => {
     setIsModalOpen(false);
   };
 
+  const updateSuggestionsDebounced = useDebounce( async () => {
+    const req = await fetch(`https://api.github.com/search/repositories?q=${encodeURIComponent(`${repoSearchTerm} in:name in:repo:owner/name`)}`, {
```
Comment (brandonroberts) [inline components/organisms/InsightPage/InsightPage.tsx:243]: Can we exclude forks and sort by most recently updated?

File: components/atoms/Search/search.tsx [hunk 3]
```diff
@@ -51,7 +51,7 @@ const Search = ({ placeholder, name, value, autoFocus, className, onSearch, sugg
     >
       <FaSearch className="text-light-slate-9" fontSize={16} onClick={handleOnSearch}/>
       <input
-        className="pl-2 w-52   placeholder:text-md  placeholder:  text-md focus:outline-none placeholder:text-light-slate-9"
+        className="pl-2 w-full placeholder:text-md  placeholder:  text-md focus:outline-none placeholder:text-light-slate-9"
```
Comment (brandonroberts) [inline components/atoms/Search/search.tsx:54]: Check for visual impact of this change in the application

**Review by brandonroberts — CHANGES_REQUESTED** [review]

File: components/atoms/Search/search.tsx (hunk 3 above)
Comment (shamimbinnur) [reply components/atoms/Search/search.tsx:54]: @brandonroberts I had to do it to make it wider on the insight page. This component has been used on `javascript/repositories` and it does not break anything visually.

File: components/organisms/InsightPage/InsightPage.tsx (hunk 2 above)
Comment (shamimbinnur) [reply components/organisms/InsightPage/InsightPage.tsx:243]: @brandonroberts  Yes, we can sort! Forks might be disabled by default. I didn't find my forked repo by searching.

Also a related link is here.
https://docs.github.com/en/search-github/searching-on-github/searching-in-forks

Comment (brandonroberts) [reply components/organisms/InsightPage/InsightPage.tsx:243]: Ok, let's sort repos by most recently updated and we don't need to add anything additional for forks

File: components/organisms/InsightPage/InsightPage.tsx (hunk 1 above)
Comment (brandonroberts) [reply components/organisms/InsightPage/InsightPage.tsx:291]: On line 291 in this PR, there are unnecessary spaces from adding a new line.

**Review by brandonroberts — APPROVED** [review]
LGTM

**Review by OgDev-01 — APPROVED** [review]
LGTM 🍕

**Pull request merged**